### PR TITLE
[instagram] Fix comments extraction

### PR DESCRIPTION
**Pull request by u-spec-png**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Fix comments extraction.

### CODE REVIEW

File: yt_dlp/extractor/instagram.py [hunk 1]
```diff
@@ -195,18 +195,19 @@ def get_count(keys, kind):
...
             comment_count = get_count(
                 ('preview_comment', 'to_comment', 'to_parent_comment'), 'comment')
 
             comments = [{
-                'author': comment.get('user', {}).get('username'),
-                'author_id': comment.get('user', {}).get('id'),
-                'id': comment.get('id'),
-                'text': comment.get('text'),
-                'timestamp': int_or_none(comment.get('created_at')),
+                'author': comment.get('node', {}).get('owner', {}).get('username'),
```
Comment (Ashish0804) [inline yt_dlp/extractor/instagram.py:204]: what's the point of a list comprehension if you are going to call `comment.get('node', {})` this multiple times?
Also it's a personal pref but the list comprehension is looking over complicated.

```py
 for comment in media.get('edge_media_to_parent_comment', {}).get('edges', []):
   comment_dict = comment.get('node', {})
   comment_text = comment_dict.get('text')
   if comment_text:
     #Append to comments using the `comment_dict` and reuse `comment_text`
```

Comment (u-spec-png) [reply yt_dlp/extractor/instagram.py]: Done.

File: yt_dlp/extractor/instagram.py [hunk 2]
```diff
@@ -200,14 +200,18 @@ def get_count(keys, kind):
...
+            comments = []
+            for comment in media.get('edge_media_to_parent_comment', {}).get('edges', []):
+                comment_dict = comment.get('node', {})
+                comment_text = comment_dict.get('text')
+                if comment_text:
+                    comments.append({
+                        'author': comment_dict.get('owner', {}).get('username'),
+                        'author_id': comment_dict.get('owner', {}).get('id'),
+                        'id': comment_dict.get('id'),
+                        'text': comment_dict.get('text'),
```
Comment (Ashish0804) [inline yt_dlp/extractor/instagram.py:212]: just do 
`'text': comment_text,`

**Review by pukkandan — CHANGES_REQUESTED** [review]
Dont use .get chains

```py
A.get(B, {}).get(C, {}).get(D)
```
should be replaced by:

```py
traverse_obj(A, (B, C, D))
```
OR
```py
try_get(A, lambda x: x[B][C][D])
```

In cases where there is only a single `.get`, leave it be

**Pull request merged**